.ci/scripts/distribution/it-java.sh

```diff
@@ -5,7 +5,7 @@ export JAVA_TOOL_OPTIONS="$JAVA_TOOL_OPTIONS -XX:MaxRAMFraction=$((LIMITS_CPU))"
 
 tmpfile=$(mktemp)
 
-mvn -o -B --fail-never -T$LIMITS_CPU -s ${MAVEN_SETTINGS_XML} verify -P skip-unstable-ci,parallel-tests -pl qa/integration-tests -pl upgrade-tests -DtestMavenId=2 -Dsurefire.rerunFailingTestsCount=7 | tee ${tmpfile}
+mvn -B --fail-never -T$LIMITS_CPU -s ${MAVEN_SETTINGS_XML} verify -P skip-unstable-ci,parallel-tests -pl qa/integration-tests -pl upgrade-tests -DtestMavenId=2 -Dsurefire.rerunFailingTestsCount=7 | tee ${tmpfile}
```

Comment:
Only the jars build by us in the `Build (Java)` stage are stashed and unstashed, we still need to download all external dependencies from nexus so this can no longer be offline

Reply:
Can't we just stash the whole `.m2` folder?

Reply:
Please don't, that will depending on the size put severe load on the Jenkins central server as per https://www.jenkins.io/doc/pipeline/steps/workflow-basic-steps/#stash-stash-some-files-to-be-used-later-in-the-build:
> This is because stashed files are archived in a compressed TAR, and with large files this demands considerable on-master resources, particularly CPU time. There's not a hard stash size limit, but between 5-100 MB you should probably consider alternatives.

Reply:
Hm, I suggested this because Optimize does it I think ;) What would you propose? Re-downloading all the artifacts is non negligible, and sometimes may cause builds to fail (which is one of the reason we decided to go offline after the initial pull in the past).

Reply:
With stashing the artifacts produced by the `Build (Java)` stage we're already a bit over the limits suggested by the documentation. To preserve the whole `.m2` folder a completely different approach needs to be taken, e.g. using a GCloud storage bucket with subfolders per branch and build number. This in turn involves having the GCloud SDK/`gsutil` tool available to the build and proper authentication etc.

Reply:
Optimize caches also not everything but just org.camunda Maven group artifacts I think.

Re-downloading is quite fine because we employ Maven caches located in the same K8s cluster as the CI jobs. They are referred using cluster-local URLs so also quite isolated and stable.

Reply:
@npepinpe do you see any other issues with not using `--offline`?

Reply:
I honestly can't remember, maybe @menski can?